tests/ui/stubbing-type-validation/type_mismatch.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// kani-flags: --harness harness --enable-unstable --enable-stubbing
+//
+//! This tests that we catch type mismatches between the stub and the original
```

Comment:
Can you please add coverage for the following:

- Return types differ.
- Wrong combination of generic parameters. Something like:
```rust
fn original<X, Y>(_x: &X, _y: &Y) -> X {}
fn stub<X, Y>(_x: &X, _y: &Y) -> Y {}
```
and
```rust
fn original<X, Y>(_x: &X, _y1: &Y, _y2: &Y) {}
fn stub<X, Y>(_x1: &X, _x2: &X, _y: &Y) {}
```

Reply:
Added!